demo-vue/src/mixins/vueStricklandMixinFactory.js

```diff
@@ -0,0 +1,178 @@
+import validate from 'strickland';
+
+const TAGNAME_SELECT = 'SELECT';
+const TAGNAME_INPUT = 'INPUT';
+const ATTRIBUTE_TYPE_CHECKBOX = 'checkbox';
+const ATTRIBUTE_TYPE_RADIO = 'radio';
+const ATTRIBUTE_MULTIPLE = 'multiple';
+
+const isSelectInput = (element) => element.tagName === TAGNAME_SELECT;
+const isCheckboxInput = (element) => element.tagName === TAGNAME_INPUT && element.type === ATTRIBUTE_TYPE_CHECKBOX;
+const isRadioInput = (element) => element.tagName === TAGNAME_INPUT && element.type === ATTRIBUTE_TYPE_RADIO;
+const shouldUseChangeEvent = (element) => isSelectInput(element) || isCheckboxInput(element) || isRadioInput(element);
+const shouldUseChangeEventOnDelay = (element) => isSelectInput(element) && element.attributes[ATTRIBUTE_MULTIPLE];
+
+export default (validator) => ({
+  data: () => ({
+    validator,
+    validation: validator.emptyResults()
+  }),
+  computed: {
+    isValid: () => this.validation && this.validation.isValid
+  },
+  methods: {
+    validateAsync (fieldResult, fieldName) {
+      this.logValidation(`strickland:validateAsync method triggered for field: '${fieldName}'.`);
+      fieldResult.validateAsync(() => this.form[fieldName])
+        .then((result) => {
+          // TODO: Do we need to check for change/race condition as per:
+          // https://strickland.io/docs/Async/RaceConditions.html
```

Comment:
It looks like you're using the automatic race condition handling:
https://strickland.io/docs/Async/AutomaticRaceConditionHandling.html

So having the catch block takes care of it.